tests/test_db.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019 Tyler Neely
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate rocksdb;
+extern crate libc;
+
+mod util;
+
+use libc::{size_t};
+
+use rocksdb::{DB, DBVector, Error, IteratorMode, Options, WriteBatch};
+use util::DBPath;
+
+#[test]
+fn test_db_vector() {
+    use std::mem;
+    let len: size_t = 4;
+    let data: *mut u8 = unsafe { mem::transmute(libc::calloc(len, mem::size_of::<u8>())) };
+    let v = unsafe { DBVector::from_c(data, len) };
+    let ctrl = [0u8, 0, 0, 0];
+    assert_eq!(&*v, &ctrl[..]);
+}
+
+#[test]
+fn external() {
+    let path = DBPath::new("_rust_rocksdb_externaltest");
+
+    {
+        let db = DB::open_default(&path).unwrap();
+        
+        let p = db.put(b"k1", b"v1111");
```

Comment:
I'd remove this binging as well.

Reply:
But I understand that this code isn't yours.

Reply:
Still, good idea to make it better.

Reply:
Fixed.